react-components/src/query/useAll3dDirectConnectionsWithProperties.ts

```diff
@@ -4,21 +4,40 @@
 
 import { type UseQueryResult, useQuery } from '@tanstack/react-query';
 import { useFdmSdk } from '../components/RevealCanvas/SDKProvider';
+import { type FdmConnectionWithNode } from '../components/CacheProvider/types';
+import { type InstanceType } from '@cognite/sdk';
+import { chunk, uniqBy } from 'lodash';
 import {
   type FdmInstanceNodeWithConnectionAndProperties,
   type FdmInstanceWithPropertiesAndTyping
-} from '../data-providers/types';
-import { type FdmConnectionWithNode } from '../components/CacheProvider/types';
-import { type InstanceType } from '@cognite/sdk';
-import { chunk, pick, uniqBy } from 'lodash';
+} from '../components/RuleBasedOutputs/types';
+import { useMemo } from 'react';
 
 export function useAll3dDirectConnectionsWithProperties(
   connectionWithNodeAndView: FdmConnectionWithNode[]
 ): UseQueryResult<FdmInstanceNodeWithConnectionAndProperties[]> {
   const fdmSdk = useFdmSdk();
 
+  const connectionKeys = useMemo(() => {
+    return connectionWithNodeAndView.map((item) => {
+      return item.connection.instance.externalId + '/' + item.connection.instance.space;
+    });
+  }, [connectionWithNodeAndView]);
+
+  const connectionWithNodeAndViewMap = useMemo(() => {
+    return new Map(
+      connectionWithNodeAndView.map((item) => {
+        return [`${item?.connection.instance.space}/${item?.connection.instance.externalId}`, item];
+      })
```

Comment:
Can you use `createFdmKey` in these two cases?

Reply:
https://github.com/cognitedata/reveal/pull/4706/commits/84b97131bc393b266b20add8adec0a2d062463c1